### PR TITLE
console: Use manifest in public dir for asset hashes

### LINKED ISSUE #1704: Use Manifest in Public Dir for Frontend Asset Hashes
#### Summary

We should write the frontend manifest to a (JSON) file in the public dir, and make The Things Stack read that instead of compiling the manifest into the binary.

#### Why do we need this?

This would allow us to compile the backend and frontend in parallel.

#### What is already there? What do you see now?

The manifest is written to `pkg/webui/manifest.go` and compiled into the binary. This means that we need to compile the frontend **before** we can compile the backend.

#### What is missing? What do you want to see?

We may be able to speed things up when we can compile them in parallel.

#### How do you propose to implement this?

- Write the manifest to a (JSON) file in the `public` dir instead of `pkg/webui/manifest.go`.
- Make `pkg/web` read that file on start and register the hashes with `pkg/webui`

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

Closes #1704

#### Changes

- Load asset hashes from a YAML manifest.
- Babel produces the yaml manifest instead of Go code that goes into the binary.

#### Notes for Reviewers

Chose YAML instead of the proposed JSON format, because it is easier to write the template.

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [X] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [X] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [X] Documentation: Relevant documentation is added or updated.
- [X] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: pkg/web/web.go [hunk 1]
```diff
@@ -168,16 +172,35 @@ func New(ctx context.Context, opts ...Option) (*Server, error) {
...
 			break
 		}
 	}
-	if staticDir != "" {
-		logger.WithFields(log.Fields("path", staticDir, "mount", options.staticMount)).Debug("Serving static assets")
+	if staticPath != "" {
+		staticDir := http.Dir(staticPath)
+		logger := logger.WithFields(log.Fields("path", staticDir, "mount", options.staticMount))
+		logger.Debug("Serving static assets")
 		s.Static(options.staticMount, staticDir, middleware.Immutable)
```
Comment (htdvisser) [inline pkg/web/web.go:186]: unrelated to this PR, but let's remove this line while we're editing this file

File: pkg/web/web.go [hunk 2]
```diff
@@ -168,16 +172,35 @@ func New(ctx context.Context, opts ...Option) (*Server, error) {
...
+			staticPath = path
 			break
 		}
 	}
-	if staticDir != "" {
-		logger.WithFields(log.Fields("path", staticDir, "mount", options.staticMount)).Debug("Serving static assets")
+	if staticPath != "" {
+		staticDir := http.Dir(staticPath)
+		logger := logger.WithFields(log.Fields("path", staticDir, "mount", options.staticMount))
+		logger.Debug("Serving static assets")
```
Comment (htdvisser) [inline pkg/web/web.go:185]: Let's put this debug line in the end.

File: pkg/web/web.go [hunk 3]
```diff
@@ -168,16 +172,35 @@ func New(ctx context.Context, opts ...Option) (*Server, error) {
...
+		// register hashed filenames
+		manifest, err := ioutil.ReadFile(filepath.Join(staticPath, "manifest.yaml"))
+		if err != nil {
+			logger.WithError(err).Warn("Failed to load manifest.yaml")
+			return s, nil
+		}
+		hashedFiles := make(map[string]string)
+		err = yaml.Unmarshal(manifest, &hashedFiles)
+		if err != nil {
+			logger.WithError(err).Warn("Corrupted manifest.yaml")
```
Comment (htdvisser) [inline pkg/web/web.go:197]: Shouldn't this error just be (wrapped and) returned?

Comment (neoaggelos) [reply pkg/web/web.go:197]: Not sure if that is what we want. Will that break development builds by chance?

If we are sure that it does not, I am fine with wrapping and returning the error.

Comment (htdvisser) [reply pkg/web/web.go:196]: I'd say that if the file exists, it must be valid yaml.

Comment (neoaggelos) [reply pkg/web/web.go]: Hmm, yeah if the file exists then it should be valid. Updated the code.

File: pkg/web/web.go (hunk 1 above)
Comment (neoaggelos) [reply pkg/web/web.go]: Removed immutable middleware

**Pull request merged**